portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,44 +35,61 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<QuotePerson> quotes;
-
   /**
-   * Add all the quotes to quotes using type QuotePerosn on start.
-   */
-  @Override
-  public void init() {
-    quotes = new ArrayList<>();
-    quotes.add(new QuotePerson("Bears. Beets. Battlestar Galactica.", "Jim Halpert"));
-    quotes.add(new QuotePerson("I'm not supersitious, but I am a little stitious.", "Michael Scott"));
-    quotes.add(new QuotePerson("The worst thing about prison was the dementors.", "Michael Scott"));
-    quotes.add(new QuotePerson("I talk a lot. so I've learned to tune myself out.", "Kelly Kapoor"));
-    quotes.add(new QuotePerson("You couldn’t handle my undivided attention.", "Dwight Schrute"));
-    quotes.add(new QuotePerson(
-        "Sometimes I'll start a sentence and I don't even know where it's going. "
-            + "I just hope I find it along the way.", "Michael Scott"));
-  }
-
-  /**
-   * For a get request, return a JSON version of a quote and person.
+   * For a get request, return a JSON version of all the comments.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    QuotePerson quote = quotes.get((int) (Math.random() * quotes.size()));
+    
+    // Get the count per page and page number from the request.
+    int commentCount = Integer.parseInt(request.getParameter("count"));
+    int pageNumber = Integer.parseInt(request.getParameter("page"));
+
+    // Prepare query and get all comments in Datastore.
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
 
-    String json = convertToJsonUsingGson(quote);
+    // Create a list of comments based on results, with a limited number of comments.
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      Comment comment = new Comment(
+        (long) entity.getKey().getId(),  
+        (String) entity.getProperty("content"), 
+        (String) entity.getProperty("author"), 
+        (long) entity.getProperty("timestamp"));
+      comments.add(comment);
+    }
 
+    List<Comment> output = new ArrayList<>();
+    if (pageNumber * commentCount > comments.size() ) {
```

Comment:
is that space ' ' after comment.size() extra?

Reply:
Yes, fixed it.